hal-core/src/addr.rs

```diff
@@ -132,32 +169,27 @@ impl Address for PAddr {
         self.0 as usize
     }
 
-    fn align_up<A: Into<usize>>(self, _align: A) -> Self {
-        unimplemented!("eliza")
+    #[inline]
+    #[cfg(target_arch = "x86_64")]
+    fn from_usize(u: usize) -> Self {
+        debug_assert!(
+            u & MASK == 0,
+            "x86_64 physical addresses may not have the 12 most significant bits set!"
+        );
```

Comment:
Might be worth `#[cfg(target_arch = "x86_64")]`-ing only this assertion, and keeping the rest of the `from_usize` impl common?